src/site/content/en/fast/codelab-optimize-third-party-javascript/index.md

```diff
@@ -0,0 +1,248 @@
+    }
+  });
+});
+
+function loadElement(element) {
+  const src = element.getAttribute('data-src');
+  element.src = src;
+}
+```
+
+Finally, add `lazy-load.js` to your document head:
+
+ ```html
+ <script src="/lazy-load.js" defer></script>
+ ```
+
+### Step 3: Reevaluate performance
+
+To see how the size and number of resources changed, open the DevTools **Network** panel and reload the page again. The **Network** panel reveals that the page made 14 requests and only 260 KB. That's a meaningful improvement!
+
+Now scroll down the page and keep an eye on the **Network** panel. When you get to the video, you should see the page trigger additional requests.
+
+<video autoplay="" loop="" muted="" playsinline="">
+    <source src="./lazy-load-3g.mp4" type="video/mp4">
+  </video>
+
+{% Aside %}
+In this example, the element is loaded when it enters the viewport and you can see it happening in the video above. You can avoid that delay and create smoother user-experience by starting to load elements a little before they enter the viewport. Use [`rootMargin`](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Creating_an_intersection_observer) property to define margins around the target element which effectively grow (or shrink) the area that triggers the `isIntersecting` change.
+{% endAside %}
+
+## Preconnect to required origins
+
+You’ve deferred non-critical Javascript and lazy-loaded the YouTube requests, so now it’s time to optimize the remaining third-party content.
+
+Adding the `rel=preconnect` attribute to a link tells the browser to establish a connection to a domain before the request for that resource is made. This attribute is best used on origins that provide resources you are certain the page needs. 
+
+The Lighthouse audit you ran in the first step suggested in **Preconnect to required origins** that you can save around 400 ms by [establishing early connections](/preconnect-and-dns-prefetch/) to staticxx.facebook.com and youtube.com:
+
+![Preconnect to required origins audit with the staticxx.facebook.com domain highlighted.](preconnect.png)
+
+Since the YouTube video is now lazy-loaded, that leaves only staticxx.facebook.com, the source of the social media sharing widget. Establishing an early connection to this domain is as simple as adding a `<link>` tag to the document's `<head>`:
+
+```html  
+  <link rel="preconnect" href="https://staticxx.facebook.com">
+```
+
+## Reevaluate performance 
+
+Here's the state of the page [after optimization](https://glitch.com/~3rd-party-optimizations). Follow the steps from the [Measure performance](#measure-performance) section of the codelab to run another Lighthouse audit. 
+
+![Lighthouse audit showing 1 second FCP and the performance score of 99.](success.png)
+
+{% Aside 'success' %}
+You should see that FCP and the performance score have improved! 🎉 The page now loads faster, consumes less data, and all without negatively affecting the user-experience. 🤝
```

Comment:
```suggestion
You should see that FCP and the performance score have improved! 🎉 The page now loads faster and consumes less data⁠—and all without negatively affecting the user-experience. 🤝
```
The "and all" construction is usually treated as an aside rather than the last item in a list.